Search parameters (#261)
meilisearch/documentation: guides/advanced_guides/search_parameters.md

| **[limit](/guides/advanced_guides/search_parameters.md#limit)**                                   | Maximum number of documents returned                            |     `20`      |
| **[attributesToRetrieve](/guides/advanced_guides/search_parameters.md#attributes-to-retrieve)**   | Attributes to display in the returned documents                 |      `*`      |
| **[attributesToCrop](/guides/advanced_guides/search_parameters.md#attributes-to-crop)**           | Attributes whose values have to be cropped                      |    `none`     |
| **[cropLength](/guides/advanced_guides/search_parameters.md#crop-length)**                        | Length used to crop field values                                |     `200`     |
| **[attributesToHighlight](/guides/advanced_guides/search_parameters.md#attributes-to-highlight)** | Attributes whose values will contain highlighted matching terms |    `none`     |
| **[filters](/guides/advanced_guides/search_parameters.md#filters)**                               | Filter queries by an attribute value                            |    `none`     |

> | **[matches](/guides/advanced_guides/search_parameters.md#matches)**                               | Defines whether the raw matches should be returned or not       |    `false`    |

Reviewer: ```suggestion
| **[matches](/guides/advanced_guides/search_parameters.md#matches)**                               | Defines whether an object that contains information about the matches should be returned or not       |    `false`    |
```

Author: Sorry, it was outdated as I replaced `shall`. I applied your changes.

:::

## Offset

A number of **documents to skip**.

`offset=<Integer>`

- `<Integer>` (Optional, positive integer, defaults to `0`)

  If the value of the parameter `offset` is _n_, _n_ first documents to skip. This is helpful for **pagination**.

#### Example

If you want to skip the **first** document, set `offset` to `1`.

```bash
$ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
      -d q=shifu \
      -d offset=1
```

## Limit

Set a **limit to the number of documents returned** by search queries.

`limit=<Integer>`

- `<Integer>` (Optional, positive integer, defaults to `20`)

  If the value of the parameter `limit` is _n_, there will be _n_ documents in the search query response. This is helpful for **pagination**.

#### Example

If you want to get only **two** documents, set `limit` to `2`.

```bash
$ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
      -d q=shifu \
      -d limit=2
```

## Attributes to retrieve

Attributes to **display** in the returned documents.

`attributesToRetrieve=<Attribute>,<Attribute>,...`

- `<Attribute>` (Optional, string, Defaults to `*`)

>   Comma-separated list of attributes that shall be in the returned documents.

Reviewer: ```suggestion
  Comma-separated list of attributes that will be in the returned documents.
```

>   Comma-separated list of attributes that shall be in the returned documents.

Reviewer: ```suggestion
  Comma-separated list of attributes whose fields will be present in the returned documents.
```

Author: Sorry, it was outdated as I replaced `shall`. I applied your changes.

#### Example

> If you want to get only the `overview` field values, set `attributesToRetrieve` to `overview`.

Reviewer: ```suggestion
If you want to get only the `overview` and `title` field and not the other fields, set `attributesToRetrieve` to `overview,title`.
```

```bash
$ curl -X GET -G 'http://localhost:7700/indexes/movies/search' \
      -d q=shifu \

>       -d attributesToRetrieve=overview

Reviewer: ```suggestion
      -d attributesToRetrieve=overview,title
```

```

## Attributes to crop

> Attributes whose values have to be cropped.

Reviewer: ```suggestion
Attributes whose values will be cropped if they contain a matched query word.
```

`attributesToCrop=<Attribute>,<Attribute>,...`

- `<Attribute>` (Optional, string, defaults to empty)

>   Comma-separated list of attributes whose values shall be cropped according to the `cropLength` value.

Reviewer: ```suggestion
  Comma-separated list of attributes whose values will be cropped if they contain a matched query word. 
  
In the case a matched query word is found, the field's value will be cropped around the first matched query word according to the `cropLength` value (default `200` see below).
```

::: tip
This is especially useful when you have to display content on the front-end in a specific way.
:::

## Crop length

> Set a **limit on the length** of field values.

Reviewer: Could you add a link to the `see attributesToCrop` to the above title?
```suggestion
Set a length for the cropping around the matching query words (see attributesToCrop).
```

`cropLength=<Integer>`

- `<Integer>` (Optional, positive integer, defaults to `200`)

>   The length used to crop field values.

Reviewer: ```suggestion
  The length used to crop field values around the first matched query word if the field's attribute is in the `attributedToCrop` list.
```

> Cropping start at the **first occurrence of the search query**.

Reviewer: ```suggestion
Cropping start at the **first occurrence of the matched query word**.
```

}
```

> In the example above, the cropped version of `overview` is 200 characters longs. There are 100 characters before `shifu` (the first match) and 100 more characters from the first letter of `shifu`.

Reviewer: ```suggestion
In the example above, the cropped version of `overview` is 200 characters longs. There are 100 characters before `shifu` (the first match) and 100 more characters starting at the first letter of `shifu`.
```

## Attributes to highlight

> Attributes whose values will contain **highlighted matching terms**.

Reviewer: ```suggestion
Attributes whose values will contain **highlighted matching query words**.
```

`attributesToHighlight=<Attribute>,<Attribute>,...`

- `<Attribute>` (Optional, string, defaults to empty)

>   Comma-separated list of attributes. Every matching string sequence in the given attribute field will be wrapped around an `<em>` tag.

Reviewer: ```suggestion
  Comma-separated list of attributes. Every matching query words in the given attribute field will be wrapped around an `<em>` tag.
```

}
```

> The parameter shall be **URL-encoded**.

Author: Oops sorry I committed this change already.

```bash
$ curl --request GET  -G 'http://localhost:7700/indexes/movies/search' \
      -d q=shifu \
      -d filters='title:Kung%20Fu%20Panda'
```

## Matches

> This setting takes a **Boolean value** (`true` or `false`) and defines whether the raw matches should be returned or not.

Reviewer: ```suggestion
This setting takes a **Boolean value** (`true` or `false`) and defines whether an object with information about the matches should be returned or not.
```

Author: ```suggestion
This setting takes a **Boolean value** (`true` or `false`) and defines whether an object that contain information about the matches should be returned or not.
```

Reviewer: Great!
